### PR TITLE
HtmlEditor: Implement converter option

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: packages/devextreme/js/__internal/ui/html_editor/m_html_editor.ts [hunk 1]
```diff
@@ -428,7 +432,11 @@ const HtmlEditor = Editor.inherit({
 
   _textChangeHandler() {
     const { value: currentValue } = this.option();
-    const convertedValue = this._deltaConverter.toHtml();
+
+    const htmlMarkup = this._deltaConverter.toHtml();
+    const convertedValue = this._htmlConverter
+      ? String(this._htmlConverter.fromHtml(htmlMarkup))
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/html_editor/m_html_editor.ts:439]: [test coverage]
There can be an error if `converter` option does not contain `fromHtml` or it's not a function

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js [hunk 2]
```diff
@@ -162,3 +166,140 @@ testModule('Custom list', {
...
+            try {
+                instance.option('value', '');
+            } catch(e) {
+                assert.ok(false, `error: ${e.message}`);
+            } finally {
+                assert.ok(true, 'there is no error');
+            }
+        });
+
+        test('coverter should be replaced in runtime', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]:
```suggestion
        test('converter option runtime change should update html converter', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js [hunk 3]
```diff
@@ -1,7 +1,11 @@
 import $ from 'jquery';
 
+import 'ui/html_editor';
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: I believe these tests should be placed not here while `converter` is just an HtmlEditor's option, not a real converter. Maybe api.tests + valueRendering.tests or like this

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js [hunk 4]
```diff
@@ -162,3 +166,140 @@ testModule('Custom list', {
...
+            instance.option('converter', secondConverter);
+            instance.option('value', 'new value');
+
+            assert.strictEqual(toHtmlFirstStub.callCount, 0);
+            assert.strictEqual(fromHtmlFirstStub.callCount, 0);
+            assert.strictEqual(toHtmlSecondStub.callCount, 1);
+            assert.strictEqual(fromHtmlSecondStub.callCount, 1);
+        });
+    });
+});
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: [test coverage]
Everywhere in these tests u check only passed functions call count, but no passed parameters and no affect for HtmlEditor's value and rendered markup.

If I remove your implementation and just call `converter.toHtml()` in option changed and `converter.fromHtml()` in `_textChangeHandler`, all tests gonna be green.

Please, improve coverage for us to be sure this functionality is working well now and cannot be broken in the future

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js [hunk 5]
```diff
@@ -162,3 +166,140 @@ testModule('Custom list', {
...
+
+        assert.strictEqual(toHtmlStub.callCount, 1);
+        assert.strictEqual(fromHtmlStub.callCount, 1);
+    });
+
+    test('toHtml and fromHtml must be called the correct number of times after the character has been entered', function(assert) {
+        const clock = sinon.useFakeTimers();
+        const toHtmlStub = sinon.stub();
+        const fromHtmlStub = sinon.stub();
+        const done = assert.async();
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: assert.async works badly together with sinon.useFakeTimers, this test can be unstable
But why in general do you need asynchrony here?

Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: done is needed to complete the test in onValueChanged. But useFakeTimers is redundant, I removed it.

File: packages/devextreme/js/__internal/ui/html_editor/m_html_editor.ts (hunk 1 above)
Comment (marker-dao) [reply packages/devextreme/js/__internal/ui/html_editor/m_html_editor.ts:439]: [e16fbce542cbcd8a29b629eac108b67c6699cd8b]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js (hunk 2 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: [e16fbce542cbcd8a29b629eac108b67c6699cd8b]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js (hunk 5 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: [e16fbce542cbcd8a29b629eac108b67c6699cd8b]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js (hunk 3 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: [bb4879dcec21e23457a3e9bab27bba29165dd7d8]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js (hunk 4 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/converters.tests.js]: [fee8d1f70c020a45f3131787aa429baa6d065d0f]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js [hunk 6]
```diff
@@ -494,6 +496,218 @@ testModule('API', moduleConfig, () => {
...
+            };
+
+            this.options = { converter };
+
+            this.createEditor();
+
+            this.instance.option('value', 'new value');
+        });
+
+        test('toHtml changed value correctly', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js]:
```suggestion
        test('toHtml changes value correctly', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js [hunk 7]
```diff
@@ -494,6 +496,218 @@ testModule('API', moduleConfig, () => {
...
+            this.options = { converter };
+
+            this.createEditor();
+
+            this.instance.option('value', 'new value');
+
+            assert.strictEqual(this.instance.option('value'), '<p>NEW VALUE</p>');
+        });
+
+        test('fromHtml changed value correctly', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js]:
```suggestion
        test('fromHtml changes value correctly', function(assert) {
```

**Review by ksercs — DISMISSED** [review]

**Review by EugeniyKiyashko — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js (hunk 6 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js]: [a1c3af4593293f483ce56d84029fb221eb8e5e31]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js (hunk 7 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js]: [a1c3af4593293f483ce56d84029fb221eb8e5e31]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**